lib/Mojo/IOLoop/Client.pm

```diff
@@ -72,18 +76,37 @@ sub _connect {
   my ($self, $args) = @_;
 
   my $handle;
-  my $address = $args->{socks_address} || $args->{address};
+  my $class;
   unless ($handle = $self->{handle} = $args->{handle}) {
-    my %options = (PeerAddr => $address, PeerPort => _port($args));
-    %options = (PeerAddrInfo => $args->{addr_info}) if $args->{addr_info};
-    $options{Blocking} = 0;
-    $options{LocalAddr} = $args->{local_address} if $args->{local_address};
+    my %options = (Blocking => 0);
+    if (defined $args->{path}) {
+      $options{Peer} = $args->{path};
+      require IO::Socket::UNIX; # load it on demand
```

Comment:
> Can't you use IO::Socket::UNIX even on Windows?

Yes, `IO::Socket::UNIX` can be loaded on Windows, the ctor call will fail later.